src/StoreApi/Utilities/OrderController.php

```diff
@@ -431,15 +431,70 @@ protected function validate_coupon_usage_limit( \WC_Coupon $coupon, \WC_Order $o
 		$coupon_usage_limit = $coupon->get_usage_limit_per_user();
 
 		if ( $coupon_usage_limit > 0 ) {
-			$data_store  = $coupon->get_data_store();
-			$usage_count = $order->get_customer_id() ? $data_store->get_usage_by_user_id( $coupon, $order->get_customer_id() ) : $data_store->get_usage_by_email( $coupon, $order->get_billing_email() );
+			$data_store = $coupon->get_data_store();
+			// First, we check a logged in customer usage count, which happens against their user id, billing email, and account email.
+			if ( $order->get_customer_id() ) {
+				$user_data = get_userdata( $order->get_customer_id() );
+				// We get usage per user id and associated emails.
+				$usage_count = $this->get_usage_per_aliases(
+					$coupon,
+					[
+						$order->get_billing_email(),
+						$order->get_customer_id(),
+						$user_data->user_email,
+					]
+				);
+			} else {
+				// Otherwise we check if the email doesn't belong to an existing user.
+				$customer_data_store = \WC_Data_Store::load( 'customer' );
+				// This will get us any user ids for this billing email.
+				$user_ids       = $customer_data_store->get_user_ids_for_billing_email( array( $order->get_billing_email() ) );
+				$emails_for_ids = array_map(
+					function( $user_id ) {
+						return get_userdata( $user_id )->user_email;
```

Comment:
Just realised this `get_userdata` and the one I added can return `false` so we need to check this.